src/ProjectTask.php

```diff
@@ -457,6 +456,9 @@ public function prepareInputForUpdate($input)
         if (isset($input['auto_percent_done']) && $input['auto_percent_done']) {
             unset($input['percent_done']);
         }
+        if (isset($input['auto_projectstates']) && $input['auto_projectstates'] == true) {
+            $input['projectstates_id'] = self::recalculateStatus($input);
```

Comment:
```suggestion
            $projectstate_id = self::recalculateStatus($input);
            if ($projectstate_id !== false) {
                $input['projectstates_id'] = $projectstate_id;
            }
```
Your new method can return `false`, in which case we shouldn't change the state value.